stage0/src/main.rs

```diff
@@ -145,6 +150,64 @@ unsafe fn branch_to_image(image: Image) -> ! {
     );
 }
 
+// Setup the MPU so that we can treat the USB RAM as normal RAM, and not as a
+// peripheral. Specifically we want to clear the `DEVICE` attributes, so that
+// we can allow unaligned access.
+//
+// NB: Portions opied from `sys/kern/src/arch/arm_m.rs:apply_memory_proteciton`
+fn apply_memory_protection(mpu: MPU) {
+    unsafe {
+        const DISABLE: u32 = 0b000;
+        const PRIVDEFENA: u32 = 0b100;
+        // From the ARMv8m MPU manual
+        //
+        // Any outstanding memory transactions must be forced to complete by
+        // executing a DMB instruction and the MPU disabled before it can be
+        // configured
+        cortex_m::asm::dmb();
+        mpu.ctrl.write(DISABLE | PRIVDEFENA);
+    }
+
+    const USB_RAM_BASE: u32 = 0x4010_0000;
+    const USB_RAM_SIZE: u32 = 0x4000;
+    const USB_RAM_REGION_NUMBER: u32 = 0;
+
+    // Subtract 32 because the `LIMIT` field in the `rlar` register range is inclusive
+    // and then enable the region (bit 0).
+    let rlar = (USB_RAM_BASE + USB_RAM_SIZE - 32) | (1 << 0);
+
+    let ap = 0b01; // read-write by any privilege level
+    let sh = 0b00; // non-shareable - we only use one core with no DMA here
+    let xn = 1; // Don't execute out of this region
+    let rbar = USB_RAM_BASE | (sh as u32) << 3 | ap << 1 | xn;
```

Comment:
hmmmm, I realize stage0 is not exactly Hubris but I wonder if we could generate this at build time as well. If we did it right we could add more regions to potentially catch stage0 touching unexpected parts of the system

Reply:
I'm up for doing this with guidance, but probably not as part of this PR.